### PR TITLE
GDScript: Fix small inconsistencies with resolve_datatype

### PR DESCRIPTION
**Pull request by vonagam**
Call `type_from_metatype` for assignable, signal parameter and typed array element type instead of simply setting `is_meta_type = false` on result of `resolve_datatype` (important for enums in terms of their built-in type).

For constants with meta types (preloads for example) `resolve_datatype` was returning non-meta result. It was not affecting anything because every time `resolve_datatype` is called its output immediately converted to non-meta one. 

Added `make_script_meta_type` while at it.

### CODE REVIEW

File: modules/gdscript/gdscript_analyzer.cpp [hunk 1]
```diff
@@ -727,10 +724,6 @@ GDScriptParser::DataType GDScriptAnalyzer::resolve_datatype(GDScriptParser::Type
 		}
 	}
 
-	if (result.builtin_type != Variant::ARRAY && p_type->container_type != nullptr) {
-		push_error("Only arrays can specify the collection element type.", p_type);
```
Comment (adamscott) [inline modules/gdscript/gdscript_analyzer.cpp:731]: Is there other types than arrays that can have a container type? What led to the removal of this part?

**Comment by adamscott** [comment]
Otherwise, good job refactoring and simplifying that part of code.

File: modules/gdscript/gdscript_analyzer.cpp (hunk 1 above)
Comment (vonagam) [reply modules/gdscript/gdscript_analyzer.cpp]: Misunderstanding on my part. Put it back now.

**Review by vnen — APPROVED** [review]
LGTM

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**